hardware_interface/include/hardware_interface/joint.hpp

```diff
@@ -0,0 +1,141 @@
+// Copyright 2020 ros2_control Development Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef HARDWARE_INTERFACE__JOINT_HPP_
+#define HARDWARE_INTERFACE__JOINT_HPP_
+
+#include <string>
+#include <vector>
+
+#include "hardware_interface/component_info.hpp"
+#include "hardware_interface/hardware_info.hpp"
+#include "hardware_interface/types/hardware_interface_return_values.hpp"
+#include "hardware_interface/visibility_control.h"
+
+namespace hardware_interface
+{
+
+/**
+ * \brief Virtual Class for the "Joint" component used as a basic build block for a robot.
+ * A joint is always 1-DoF and can have one or more interfaces (e.g., position, velocity, etc.)
+ * A joint has to be able to receive command(s) and optionally can provide its state(s).
+ * The lists of command and state interfaces define this.
+ */
+class Joint
+{
+public:
+  HARDWARE_INTERFACE_PUBLIC
+  Joint() = default;
+
+  HARDWARE_INTERFACE_PUBLIC
+  virtual
+  ~Joint() = default;
+
+  /**
+   * @brief Configure joint based on the description in the robot's URDF file.
```

Comment:
doxygen style, same for the rest of this file.

Reply:
s**** I forgot find/replace before push...